synse/http_integration_test.go

```diff
@@ -0,0 +1,551 @@
+	assert.NoError(t, err)
+
+	opts := scheme.ScanOptions{}
+	devices, err := client.Scan(opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 22, len(devices))
+
+	for _, device := range devices {
+		assert.NotEmpty(t, device.ID)
+		assert.NotEmpty(t, device.Info)
+		assert.NotEmpty(t, device.Type)
+		assert.NotEmpty(t, device.Plugin)
+		assert.NotEmpty(t, device.Tags)
+
+		// only led devices have alias
+		if device.Type == "led" {
+			assert.NotEmpty(t, device.Alias)
+		} else {
+			assert.Empty(t, device.Alias)
+		}
+	}
+}
+
+func TestIntegration_Tags(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	opts := scheme.TagsOptions{}
+	tags, err := client.Tags(opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 10, len(tags))
+
+	for _, tag := range tags {
+		assert.NotEmpty(t, tag)
+	}
+}
+
+func TestIntegration_Info(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	opts := scheme.ScanOptions{}
+	devices, err := client.Scan(opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 22, len(devices))
+
+	for _, device := range devices {
+		assert.NotEmpty(t, device)
+	}
+
+	for _, device := range devices {
+		assert.NotEmpty(t, device)
+
+		info, err := client.Info(device.ID)
+		assert.NoError(t, err)
+		assert.NotEmpty(t, info.Timestamp)
+		assert.NotEmpty(t, info.ID)
+		assert.NotEmpty(t, info.Type)
+		assert.NotEmpty(t, info.Plugin)
+		assert.NotEmpty(t, info.Capabilities.Mode)
+		assert.NotEmpty(t, info.Tags)
+		assert.NotEmpty(t, info.Metadata)
+		assert.Empty(t, info.Capabilities.Write.Actions)
+
+		// TODO - add sort_index to the scheme
+
+		// only led devices have alias
+		if info.Type == "led" {
+			assert.NotEmpty(t, info.Alias)
+		} else {
+			assert.Empty(t, info.Alias)
+		}
+
+		for _, output := range info.Outputs {
+			assert.NotEmpty(t, output.Name)
+			// assert.NotEmpty(t, output.Type) // FIXME - only airflow types are empty?
+
+			// led and lock devices don't produce unit and precision output
+			if output.Name == "state" || output.Name == "color" || output.Name == "status" {
+				assert.Empty(t, output.Unit.Name)
+				assert.Empty(t, output.Unit.Symbol)
+				assert.Empty(t, output.Precision)
+			} else {
+				assert.NotEmpty(t, output.Unit.Name)
+				assert.NotEmpty(t, output.Unit.Symbol)
+				assert.NotEmpty(t, output.Precision)
+			}
+
+			assert.Empty(t, output.ScalingFactor)
+		}
+	}
+}
+
+func TestIntegration_Read(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	opts := scheme.ReadOptions{}
+	devices, err := client.Read(opts)
```

Comment:
nitpick, but we aren't getting devices back, so the variable name can be misleading -- could change this to 'readings'